### PR TITLE
working on cosmos mode

### PR DESCRIPTION
**Pull request by hannydevelop**

### CODE REVIEW

File: src/commands/cosmos_mode.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            let client = SignerMiddleware::new(client, wallet.clone());
+            let client = Arc::new(client);
+            let mongo = config.mongo.clone();
+
+            let name = &config
+            .keys
+            .rebalancer_key;
+            let cosmos_key = config.load_deep_space_key(name.clone());
+
+            let poller = Poller::new(&cellar, client, &mongo, &cosmos_key)
```
Comment (zmanian) [inline src/commands/cosmos_mode.rs]: So I think hen the poller gets constructed an argument should be passed in determining if it's in single signer or cosmos mode

File: src/collector/poller.rs [hunk 2]
```diff
@@ -109,6 +149,28 @@ impl<T: 'static + Middleware> Poller<T> {
             Ok(())
         } else {
             tick_info.reverse();
+            // TODO(Levi) needs to be initialized (should it be derived from the cosmos_key??)
+            let delegate_cosmos_address =
```
Comment (zmanian) [inline src/collector/poller.rs]: Should be derived from the cosmos key that gets passed to the poller constructor

**Review by zmanian — CHANGES_REQUESTED** [review]

File: src/collector/poller.rs (hunk 2 above)
Comment (hannydevelop) [reply src/collector/poller.rs]: Done https://github.com/PeggyJV/cellar_rebalancer_rs/blob/c3f93f33563adcd63ff450bf84384f464b67a8ec/src/collector/poller.rs#L155

**Comment by hannydevelop** [comment]
I'll have to start this PR over and close up this, that's my goal today.

**Pull request closed without merging**